src/main/java/ru/skypro/homework/security/SecurityUser.java

```diff
@@ -0,0 +1,74 @@
+package ru.skypro.homework.security;
+
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.RequestScope;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+@Component
+@RequestScope(proxyMode = ScopedProxyMode.TARGET_CLASS)
+public class SecurityUser implements UserDetails {
+
+    private SecurityUserDto securityUserDto;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return Optional.ofNullable(securityUserDto)
+                .map(SecurityUserDto::getRole)
+                .map(Enum::name)
+                .map(role -> "ROLE_" + role)
+                .map(SimpleGrantedAuthority::new)
+                .map(Collections::singleton)
+                .orElse(null);
```

Comment:
```suggestion
                .orElseGet(Collections::emptySet);
```
лучше пустую коллекцию вернуть

Reply:
Спасибо.